dashboard/src/components/DeploymentFormBody/BasicDeploymentForm/CustomFormParam.tsx

```diff
@@ -14,7 +14,14 @@ export default function CustomFormComponentLoader({
   handleBasicFormParamChange,
 }: ICustomParamProps) {
   // Fetches the custom-component bundle served by the dashboard nginx
-  const url = `${window.location.origin}/custom_components.js`;
+
+  const {
+    config: { remoteComponentsUrl },
+  } = useSelector((state: IStoreState) => state);
+
+  const url = remoteComponentsUrl
```

Comment:
As this URL is a Kubeapps' admin responsibility, it's ok accepting any value, but, have you consider these scenarios?

- Kubeapps running with HTTPS trying to load a custom component served via HTTP.
- Kubeapps running in a domain totally different from the one where the custom component is served (i.e., cross-origin requests).

I'm just wondering if adding this remote URL loading would also imply some changes in the headers (CSP, CORS, etc) used in the nginx server.

Reply:
So funny enough I ran into this! I was doing some testing with a WDS that I would load some components from, I needed to add these headers to the WDS to allow the request from my kubeapps frontend
```
headers: {
      'Access-Control-Allow-Origin': '*',
      'Access-Control-Allow-Methods': 'GET, POST, PUT, DELETE, PATCH, OPTIONS',
      'Access-Control-Allow-Headers':
        'X-Requested-With, content-type, Authorization'
    }
```